File: mediator/golang/waiter.go
package main

// OurWaiter extends Waiter and coordinates specific chefs
type OurWaiter struct {
	SoupChef     *SoupChef
	SandwichChef *SandwichChef
}

func (w *OurWaiter) InformChef(chef Chef, message string) {
	chef.ReceiveOrder(message)
}

File: mediator/rust/mediator/src/waiter.rs
use crate::chef::Chef;

/// The Mediator trait
pub trait Waiter {
    fn inform_chef(&self, chef: &dyn Chef, message: &str);
}

/// Extended Waiter implementation
pub struct OurWaiter;

impl Waiter for OurWaiter {
    fn inform_chef(&self, chef: &dyn Chef, message: &str) {
        println!("Waiter delivering message: {}", message);
        chef.receive_order(message);
    }
}
